tensorrt_llm/_torch/speculative/eagle3.py

```diff
@@ -90,8 +91,8 @@ class Eagle3SpecMetadata(SpecMetadata):
     eagle3_resource_manager: Optional[Eagle3ResourceManager] = None
 
     def __post_init__(self):
-        if self.num_layers == 1:
-            self.layers_to_capture = (0, )
+        if self.num_layers == 1 or self.num_capture_layers == 1:
+            self.layers_to_capture = (self.num_layers - 1, )
```

Comment:
Can we just get rid of `num_capture_layers`?  It should be equal to `len(layers_to_capture)`. Feel free to add a read-only `@property` that derives it from this field for convenience though.